### ISSUE TITLE
Add support for other table formats (esp. as in pandoc)

### ISSUE THREAD

**Issue by Jmuccigr** [hoedown/hoedown#175]
_(Apologies if such requests should be discussed elsewhere.)_

Has there been any thought given to adding support for other table formats besides pipe? I'm especially interested in grid tables, because they can support block elements, like lists. Pandoc supports grid and others and it would be great to use them for greater flexibility in table-making.

**Comment by jmendeth** [comment]
At Hoedown, the philosophy has usually been to implement syntax extensions that are widely used and popular in the Markdown community. Footnotes would be an example of such an extension.

When it comes to tables, pipe syntax is the clear winner, and [Kramdown table syntax](kramdown.gettalong.org/syntax.html#tables) a good standard for it. So that syntax will be present in the upcoming version.

However, I agree with you that pipe syntax has limitations so we it can be a good idea to implement Pandoc's grid syntax as well. I'll leave this issue open.

**Comment by Jmuccigr** [comment]
Looking at the Kramdown syntax [here](http://kramdown.gettalong.org/syntax.html#tables), I think I see that it already supports grid tables, but not block elements inside cells, since it says:
> Table cells can only contain a single line of text, no multi-line text is supported. The text of a table cell is parsed as span-level elements.

It also looks like the header is optional, which is nice too. Both of those things are desirable.

**Comment by jmendeth** [comment]
> I think I see that it already supports grid tables

No, Kramdown is using pipe syntax, because every line of text is a row (regardless of ruler placement), columns are aligned based on colons in the first ruler, and cell content is span-level instead of block-level, as you said.

Both syntaxes collide in a way that it's easy to make tables that are valid under both syntaxes.
We'll have to work around that it we want to implement grid syntax in Hoedown...

**Comment by Jmuccigr** [comment]
I meant your second point, that it's already parsing what are valid grid tables (in at least some way).
